aws/rust-runtime/aws-runtime/src/auth.rs

```diff
@@ -25,6 +31,53 @@ pub mod sigv4 {
     /// Auth scheme ID for SigV4.
     pub const SCHEME_ID: AuthSchemeId = AuthSchemeId::new("sigv4");
 
+    struct EndpointAuthSchemeConfig {
+        signing_region_override: Option<SigningRegion>,
+        signing_service_override: Option<SigningService>,
+    }
+
+    #[derive(Debug)]
+    enum SigV4SigningError {
+        MissingOperationSigningConfig,
+        MissingSigningRegion,
+        MissingSigningService,
+        WrongIdentityType(Identity),
+        BadTypeInEndpointAuthSchemeConfig(&'static str),
+    }
+
+    impl fmt::Display for SigV4SigningError {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            use SigV4SigningError::*;
+            let mut w = |s| f.write_str(s);
+            match self {
+                MissingOperationSigningConfig => w("missing operation signing config for SigV4"),
+                MissingSigningRegion => w("missing signing region for SigV4 signing"),
+                MissingSigningService => w("missing signing service for SigV4 signing"),
+                WrongIdentityType(identity) => {
+                    write!(f, "wrong identity type for SigV4: {identity:?}")
+                }
+                BadTypeInEndpointAuthSchemeConfig(field_name) => {
+                    write!(
+                        f,
+                        "unexpected type for `{field_name}` in endpoint auth scheme config",
+                    )
+                }
+            }
+        }
+    }
+
+    impl StdError for SigV4SigningError {
+        fn source(&self) -> Option<&(dyn StdError + 'static)> {
+            match self {
+                Self::MissingOperationSigningConfig => None,
+                Self::MissingSigningRegion => None,
+                Self::MissingSigningService => None,
+                Self::WrongIdentityType(_) => None,
+                Self::BadTypeInEndpointAuthSchemeConfig(_) => None,
+            }
```

Comment:
imo, just do:
```suggestion
            // SigV4 errors never have a source.
            None
```

Reply:
The errors RFC specifically calls out not doing this 😄